File: src/Controls/Vns.Erp.Core.Controls/UserControl/C_SoChungTu.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using Vns.Erp.Core.Admin.Domain;
using Vns.Erp.Core.Admin.Service.Interface;

namespace Vns.Erp.Core.Controls.UserControl
{
    public partial class C_SoChungTu
    {

        #region "Properties"

        private string _PREPIX;

        private string _SO_CHUNG_TU;
        #region "Property"
        private IHtLoaichungtuService _HtLoaichungtuService;
        public IHtLoaichungtuService HtLoaichungtuService
        {
            get { return _HtLoaichungtuService; }
            set { _HtLoaichungtuService = value; }
        }
        #endregion

        public string SO_CHUNG_TU
        {
            get { return _SO_CHUNG_TU; }
            set { _SO_CHUNG_TU = value; }
        }

        public string SoChungTu
        {
            get
            {
                if (String.IsNullOrEmpty(_PREPIX)) 
                    return _SO_CHUNG_TU;
                else
                    return String.Format(_PREPIX, _SO_CHUNG_TU);
            }
        }

        public string PREFIX
        {
            get { return _PREPIX; }
            set { _PREPIX = value; }
        }

        public override string Text
        {
            get { return txtSO_CT.Text; }
            set { txtSO_CT.Text = value; }
        }
        #endregion

        #region "Events"

        public void r_SoCTPREFIX(Guid p_LOAICHUNGTU_ID, decimal p_THANG, decimal p_NAM, Guid p_DONVI_ID)
        {
            if ((_HtLoaichungtuService == null))
            {
                _HtLoaichungtuService = (IHtLoaichungtuService)ObjectFactory.GetObject("HtLoaichungtuService");
            }
            List<string> lst = _HtLoaichungtuService.GetSoCT_prefix(p_LOAICHUNGTU_ID, p_THANG, p_NAM, p_DONVI_ID);
            if (lst != null)
            {
                _PREPIX = lst[0];
                _SO_CHUNG_TU = lst[1];
            }
        }

        public void SetReadOnly()
        {
            txtSO_CT.Properties.ReadOnly = true;
        }

        private void C_SoChungTu_Load(System.Object sender, System.EventArgs e)
        {
            try
            {
                if (this.DesignMode)
                {
                    return;
                }
                //txtSO_CT.Text = _SO_CHUNG_TU
            }
            catch (Exception ex)
            {
                FormGlobals.Message_Error(ex);
            }
        }
        public C_SoChungTu()
        {
            InitializeComponent();
        }
        #endregion

    }
}

File: src/Accounting/Vns.Erp.Core.Accounting.VbAccountingGUI/UserControl/C_SoChungTu.vb
Imports Vns.Erp.Core.Admin.Domain
Imports Vns.Erp.Core.Admin.Service.Interface

Public Class C_SoChungTu

#Region "Properties"

    Private _PREPIX As String
    Private _SO_CHUNG_TU As String

#Region "Property"
    Private _HtLoaichungtuService As IHtLoaichungtuService
    Public Property HtLoaichungtuService As IHtLoaichungtuService
        Get
            Return _HtLoaichungtuService
        End Get
        Set(ByVal value As IHtLoaichungtuService)
            _HtLoaichungtuService = value
        End Set
    End Property
#End Region

    Public Property SO_CHUNG_TU() As String
        Get
            Return _SO_CHUNG_TU
        End Get
        Set(ByVal value As String)
            _SO_CHUNG_TU = value
        End Set
    End Property

    Public Property PREFIX() As String
        Get
            Return _PREPIX
        End Get
        Set(ByVal value As String)
            _PREPIX = value
        End Set
    End Property

    Public Overrides Property Text() As String
        Get
            Return txtSO_CT.Text
        End Get
        Set(ByVal value As String)
            txtSO_CT.Text = value
        End Set
    End Property
#End Region

#Region "Events"

    Public Sub r_SoCTPREFIX(ByVal p_LOAICHUNGTU_ID As Guid, ByVal p_THANG As Decimal, ByVal p_NAM As Decimal, ByVal p_DONVI_ID As Guid)
        If (_HtLoaichungtuService Is Nothing) Then
            _HtLoaichungtuService = CType(ObjectFactory.GetObject("HtLoaichungtuService"), IHtLoaichungtuService)
        End If
        Dim lst As List(Of String) = _HtLoaichungtuService.GetSoCT_prefix(p_LOAICHUNGTU_ID, p_THANG, p_NAM, p_DONVI_ID)
        If lst IsNot Nothing Then
            _PREPIX = lst(0)
            _SO_CHUNG_TU = lst(1)
        End If
    End Sub

    Private Sub C_SoChungTu_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Try
            If Me.DesignMode Then
                Return
            End If
            'txtSO_CT.Text = _SO_CHUNG_TU
        Catch ex As Exception
            Message_Error(ex)
        End Try
    End Sub
#End Region

End Class
